imperial_coldfront_plugin/views.py

```diff
@@ -36,11 +37,74 @@
     check_group_owner_or_superuser,
     user_already_has_hpc_access,
     user_eligible_for_hpc_access,
+    user_eligible_to_be_pi,
 )
 
 User = get_user_model()
 
 
+@login_required
+def research_group_terms_view(request: HttpRequest) -> HttpResponse:
+    """View for accepting T&Cs and creating a ResearchGroup.
+
+    TODO: Verify if superusers should be able to create research groups.
+
+    Args:
+        request: The Http request including the user information.
+
+    Returns:
+        The relevant Http response, depending on the permissions and the type of
+        request.
+    """
+    graph_client = get_graph_api_client()
+    user_profile = graph_client.user_profile(request.user.username)
+    if not user_eligible_to_be_pi(user_profile) and not request.user.is_superuser:
+        return HttpResponseForbidden("You are not allowed to create a research group.")
+    elif GroupMembership.objects.filter(member=request.user).exists():
+        return HttpResponseForbidden(
+            "You cannot create a research group while being a member of another one.",
+        )
+
+    if request.method == "POST":
+        form = TermsAndConditionsForm(request.POST)  # use TermsAndConditionsForm
+        if form.is_valid():
+            # Autogenerate name
+            group_name = f"Research Group {request.user.username}"
+
+            # If the group already exist, we just use that one
+            if (query := ResearchGroup.objects.filter(owner=request.user)).exists():
+                group = query.first()
+                messages.success(
+                    request,
+                    f"A research group owned by '{request.user}' already exist.",
+                )
+            else:
+                gid = generate_unique_gid()
+                group = ResearchGroup.objects.create(
+                    owner=request.user, gid=gid, name=group_name
+                )
+                messages.success(request, "Research group created successfully.")
```

Comment:
Not a big deal but could maybe use [get_or_create](https://docs.djangoproject.com/en/5.1/ref/models/querysets/#get-or-create) here. Or a `Research.objects.filter` with a try except block.

Reply:
I had forgotten about get_or_create... :) Much neater, now.